lib/jsdom/living/nodes/HTMLCanvasElement-impl.js

```diff
@@ -5,33 +5,67 @@
 const HTMLElementImpl = require("./HTMLElement-impl").implementation;
 
 const notImplemented = require("../../browser/not-implemented");
+const idlUtils = require("../generated/utils");
+
 let Canvas;
 try {
   Canvas = require("canvas");
+  if (typeof Canvas !== "function") {
+    // In browserify, the require will succeed but return an empty object
+    Canvas = null;
+  }
 } catch (e) {
   // Do nothing
 }
 
 class HTMLCanvasElementImpl extends HTMLElementImpl {
-  constructor(args, privateData) {
-    super(args, privateData);
 
-    if (typeof Canvas === "function") { // in browserify, the require will succeed but return an empty object
-      this._nodeCanvas = new Canvas(this.width, this.height);
+  _attrModified(name, value) {
+    if (this._canvas && (name === "width" || name === "height")) {
+      this._canvas[name] = parseInt(value);
     }
+
+    return super._attrModified.apply(this, arguments);
   }
 
-  _attrModified(name) {
-    if ((name === "width" || name === "height") && this._nodeCanvas) {
-      this._nodeCanvas = new Canvas(this.width, this.height);
+  _getCanvas() {
+    if (Canvas && !this._canvas) {
+      this._canvas = new Canvas(this.width, this.height);
     }
-
-    return super._attrModified.apply(this, arguments);
+    return this._canvas;
   }
 
   getContext(contextId) {
-    if (this._nodeCanvas) {
-      return this._nodeCanvas.getContext(contextId) || null;
+    // We need to wrap the methods that receive an image or canvas object
+    // (luckily, always as the first argument), so that these objects can be
+    // unwrapped an the expected types passed.
+    function wrap(ctx, name) {
```

Comment:
While merging I am going to move this function to the top-level of the module so that we don't re-create it every time someone calls getContext.